virttest/env_process.py

```diff
@@ -560,6 +560,25 @@ def preprocess(test, params, env):
     :param env: The environment (a dict-like object).
     """
     error_context.context("preprocessing")
+
+    # For KVM to work in Power8 systems we need to have SMT=off
+    # and it needs to be done as root, here we do a check whether
+    # we satisfy that condition, if not try to make it off
+    # otherwise throw TestError with respective error message
+    cmd = "grep cpu /proc/cpuinfo | awk '{print $3}' | head -n 1"
```

Comment:
@sathnaga  it looks enable/disable SMT is  prepare test step for "Power8", so I think write a function like "disable_smt" in test_setup module and call it in env_process module looks more reasonable. And raise "exceptions.TestSetupFail" exceptions looks more suitable than "TestError" exception when disable MST fail;